### PR TITLE
Fix notification opt-out checking on Samsung devices

### PR DESCRIPTION
**Pull request by tjleing**
On Samsung devices running Android 12, there was a strange bug impacting uninstalling and then reinstalling an app using Pinpoint -- the library would report that push notifications were disabled when they weren't.  Using the more standard API `areNotificationsEnabled()` (supported under API versions 24 and up) as opposed to the existing workaround should fix the issue.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by gpanshu — APPROVED** [review]

File: aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/targeting/notification/NotificationClientBase.java [hunk 1]
```diff
@@ -975,6 +976,10 @@ boolean areAppNotificationsEnabledOnPlatform() {
             return true;
         }
 
+        if (android.os.Build.VERSION.SDK_INT >= ANDROID_NOUGAT) {
```
Comment (sktimalsina) [inline aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/targeting/notification/NotificationClientBase.java:979]: Can we update the documentation on behavior for API 24+ as well?

**Review by sktimalsina — APPROVED** [review]
LGTM! Feel free to merge after updating the documentation.

File: aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/targeting/notification/NotificationClientBase.java [hunk 2]
```diff
@@ -957,6 +958,8 @@ public final NotificationClient.PushResult handleCampaignPush(NotificationDetail
      * notifications was a feature added on devices supporting API Level 16 and
      * above, so devices from API level 16 to 18 will return true from this
      * method even when local notifications have been disabled for the app.
+     * Also, at API level 24 and higher, we use the more consistent API
+     * `areNotificationsAEnabled()` to fix an issue we were seeing with Samsung devices.
```
Comment (sktimalsina) [inline aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/targeting/notification/NotificationClientBase.java:962]: NotificationManagerCompat#areNotificationsEnabled()

**Comment by tjleing** [comment]
Unit tests are failing because `NotificationManagerCompat` isn't initializing correctly in a mock environment.

**Pull request merged**